pkg/storage/kubernetes.go

```diff
@@ -227,3 +239,197 @@ func (t *temporaryError) Temporary() bool {
 type temporary interface {
 	Temporary() bool
 }
+
+// newLeaderElector creates a new leaderelection.LeaderElector and associated
+// channels by which to observe elections and depositions.
+func newLeaderElector(clientset *kubernetes.Clientset, namespace string, podNamespace string, podID string) (*leaderelection.LeaderElector, chan struct{}, chan struct{}) {
+	//log.WithField("context", "leaderelection")
+	// leaderOK will block gRPC startup until it's closed.
+	leaderOK := make(chan struct{})
+	// deposed is closed by the leader election callback when
+	// we are deposed as leader so that we can clean up.
+	deposed := make(chan struct{})
+
+	var rl = &resourcelock.LeaseLock{
+		LeaseMeta: metav1.ObjectMeta{
+			Name:      "whereabouts",
+			Namespace: namespace,
+		},
+		Client: clientset.CoordinationV1(),
+		LockConfig: resourcelock.ResourceLockConfig{
+			Identity: fmt.Sprintf("%s/%s", podNamespace, podID),
+		},
+	}
+
+	// Make the leader elector, ready to be used in the Workgroup.
+	le, err := leaderelection.NewLeaderElector(leaderelection.LeaderElectionConfig{
+		Lock:          rl,
+		LeaseDuration: 1500 * time.Millisecond,
+		RenewDeadline: 1000 * time.Millisecond,
+		RetryPeriod:   500 * time.Millisecond,
```

Comment:
Can we make it configurable from CNI json config? It might be key for performance tuning.

Reply:
Added changes for configurable parameters for lease/renew/retry durations